spec/acceptance/producer_spec.rb

```diff
@@ -130,14 +130,31 @@ class { 'kafka::producer':
         apply_manifest(pp, catch_changes: true)
       end
 
-      describe file('/etc/init.d/kafka-producer') do
+      describe file('/etc/init.d/kafka-producer'), if: (fact('operatingsystemmajrelease') =~ %r{(5|6)} && fact('osfamily') == 'RedHat') do
```

Comment:
kafka-producer service was missing the same os-specific spec tests that existed for the other services.